### PR TITLE
Derive exported EndpointSlices from K8s EndpointSlices instead of Endpoints

### LINKED ISSUE #755: EndpointSlices generated by lighthouse agent doesn't capture all endpoint conditions
**What happened**:
The endpointslices generated by submariner-operator don't capture all the conditions associated with an endpoint. Only ready condition is captured while serving and terminating are not captured.
**What you expected to happen**:
All the conditions associated with an endpoint should be captured. All conditions including ready, serving and terminating.
**How to reproduce it (as minimally and precisely as possible)**:
Check endpointslices generated for any exported service and compare it with the endpointslices generated by endpointscontroller.
**Anything else we need to know?**:

**Environment**:
- Diagnose information (use `subctl diagnose all`):
- Gather information (use `subctl gather`)
- Cloud provider or hardware configuration:
- Install tools:
- Others:
[endpointcontroller-slices.txt](https://github.com/submariner-io/submariner-operator/files/8663348/endpointcontroller-slices.txt)
[submariner-slices.txt](https://github.com/submariner-io/submariner-operator/files/8663349/submariner-slices.txt)

### LINKED ISSUE #1022: Flaky LH e2e test failure
**What happened**:

The following e2e test-case is occasionally failing in CI.

```
[Fail] [discovery] Test Headless Service Discovery Across Clusters when the number of active pods backing a service changes [It] should resolve the local pod IPs 
/go/src/github.com/submariner-io/lighthouse/vendor/github.com/submariner-io/shipyard/test/e2e/framework/framework.go:562
```

```
[discovery] Test Headless Service Discovery Across Clusters when the number of active pods backing a service changes 
  should resolve the local pod IPs
  /go/src/github.com/submariner-io/lighthouse/test/e2e/discovery/headless_services.go:57
STEP: Creating namespace objects with basename "discovery"
STEP: Generated namespace "e2e-tests-discovery-4r8dt" in cluster "cluster1" to execute the tests in
STEP: Creating namespace "e2e-tests-discovery-4r8dt" in cluster "cluster2"
STEP: Creating namespace "e2e-tests-discovery-4r8dt" in cluster "cluster3"
STEP: Creating an Nginx Deployment on "cluster1"
STEP: Setting Nginx deployment replicas to 3 in cluster "cluster1"
STEP: Creating a Nginx Headless Service on "cluster1"
STEP: Creating serviceExport nginx-headless.e2e-tests-discovery-4r8dt on "cluster1"
STEP: Retrieving ServiceExport nginx-headless.e2e-tests-discovery-4r8dt on "cluster1"
STEP: Creating a Netshoot Deployment on "cluster1"
STEP: Executing "dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local" to verify IPs [10.129.0.24 10.129.0.23 10.129.0.22] for service "nginx-headless" "are" discoverable
Dec 22 01:43:20.820: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "10.129.0.22\n10.129.0.23\n10.129.0.24"
STEP: Setting Nginx deployment replicas to 0 in cluster "cluster1"
STEP: Executing "dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local" to verify IPs [] for service "nginx-headless" "are not" discoverable
Dec 22 01:43:55.952: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are not ""
STEP: Setting Nginx deployment replicas to 2 in cluster "cluster1"
STEP: Executing "dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local" to verify IPs [10.129.0.27 10.129.0.26] for service "nginx-headless" "are" discoverable
Dec 22 01:44:01.063: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:06.147: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:11.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:16.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:21.149: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:26.147: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:31.147: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:36.147: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:41.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:46.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:51.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:44:56.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:01.147: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:06.149: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:11.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:16.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:21.149: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:26.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:31.147: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:36.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:41.149: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:46.149: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:51.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:45:56.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:01.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:06.149: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:11.149: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:16.149: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:21.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:26.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:31.149: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:36.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:41.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:46.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:51.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:46:56.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:47:01.150: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:47:06.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:47:11.151: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
Dec 22 01:47:11.250: INFO: ExecWithOptions &{Command:[dig +short nginx-headless.e2e-tests-discovery-4r8dt.svc.clusterset.local] Namespace:e2e-tests-discovery-4r8dt PodName:netshootbknzn-8594567f-5dr6x ContainerName:netshoot Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:false}
STEP: Validating that dig result are "242.254.1.252\n10.129.0.27"
STEP: Deleting namespace "e2e-tests-discovery-4r8dt" on cluster "cluster1"
STEP: Deleting namespace "e2e-tests-discovery-4r8dt" on cluster "cluster2"
STEP: Deleting namespace "e2e-tests-discovery-4r8dt" on cluster "cluster3"
STEP: Retrieving EndpointSlices for "" in ns "e2e-tests-discovery-4r8dt" on "cluster2"
STEP: Retrieving EndpointSlices for "" in ns "e2e-tests-discovery-4r8dt" on "cluster1"

• Failure [255.658 seconds]
[discovery] Test Headless Service Discovery Across Clusters
/go/src/github.com/submariner-io/lighthouse/test/e2e/discovery/headless_services.go:37
  when the number of active pods backing a service changes
  /go/src/github.com/submariner-io/lighthouse/test/e2e/discovery/headless_services.go:52
    should resolve the local pod IPs [It]
    /go/src/github.com/submariner-io/lighthouse/test/e2e/discovery/headless_services.go:57

    Failed to  service IP verification. expected execution result "242.254.1.252\n10.129.0.27" to contain "10.129.0.26"
    Unexpected error:
        <*errors.errorString | 0xc00019a0f0>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    occurred

    /go/src/github.com/submariner-io/lighthouse/vendor/github.com/submariner-io/shipyard/test/e2e/framework/framework.go:562

    Full Stack Trace
    github.com/submariner-io/shipyard/test/e2e/framework.AwaitUntil({0x18a9ef1?, 0x3c?}, 0x0?, 0x0?)
    	/go/src/github.com/submariner-io/lighthouse/vendor/github.com/submariner-io/shipyard/test/e2e/framework/framework.go:562 +0xd4
    github.com/submariner-io/lighthouse/test/e2e/framework.(*Framework).VerifyIPsWithDig(0xc0003afa00, 0x0, 0xc000004c80, 0xc000480fc0, {0xc000407c20?, 0x2, 0x2}, {0x26df750, 0x1, 0x1}, ...)
    	/go/src/github.com/submariner-io/lighthouse/test/e2e/framework/framework.go:639 +0x590
    github.com/submariner-io/lighthouse/test/e2e/discovery.RunHeadlessPodsAvailabilityTestLocal(0x0?)
    	/go/src/github.com/submariner-io/lighthouse/test/e2e/discovery/headless_services.go:226 +0x3a9
    github.com/submariner-io/lighthouse/test/e2e/discovery.glob..func1.3.2()
    	/go/src/github.com/submariner-io/lighthouse/test/e2e/discovery/headless_services.go:58 +0x1d
    github.com/submariner-io/shipyard/test/e2e.RunE2ETests(0x48f9b7?)
    	/go/src/github.com/submariner-io/lighthouse/vendor/github.com/submariner-io/shipyard/test/e2e/e2e.go:99 +0x136
    github.com/submariner-io/lighthouse/test/e2e_test.TestE2E(0x0?)
    	/go/src/github.com/submariner-io/lighthouse/test/e2e/e2e_test.go:28 +0x19
    testing.tRunner(0xc0003bad00, 0x1978070)
    	/usr/lib/golang/src/testing/testing.go:1446 +0x10b
    created by testing.(*T).Run
    	/usr/lib/golang/src/testing/testing.go:1493 +0x35f
------------------------------
```

On a high level, the test-case does the following.
1. Creates an nginx deployment with 3 replicas.
2. Creates a Headless Service in front of the nginx deployment
3. Exports the Headless Service
4. Creates a netshoot client in the same cluster and tries to connect to the headless service.
5. Verifies if Lighthouse successfully resolves the DNS for the exported headless service. 
6. The test-case modifies the replica count to 0 and re-verifies that LH does not resolve it any more.
7. So far all good.
8. It then re-configures the replica count to 2 and tries to resolve the IPs. At this stage, one of the IPs returned by LH is a globalIP instead of local-ip, because of this, the test-case is marked as failed.

We have to investigate why LH is returning a globalIP instead of local-ip for one of the Headless service pods.

Sample failures:
https://github.com/submariner-io/lighthouse/actions/runs/3754156273/jobs/6378122029
https://github.com/submariner-io/lighthouse/actions/runs/3736189849/jobs/6340228907

### PR DESCRIPTION
**Pull request by tpantelis**
See [this comment](https://github.com/submariner-io/lighthouse/issues/1022#issuecomment-1504072763) for reasoning.

See individual commits for details.

Fixes #1022
Fixes #755

### CODE REVIEW

File: coredns/resolver/resolver_suite_test.go [hunk 1]
```diff
@@ -302,7 +303,7 @@ func newEndpointSlice(namespace, name, clusterID string, ports []mcsv1a1.Service
 
 	return &discovery.EndpointSlice{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      name + "-" + namespace + "-" + clusterID,
+			Name:      fmt.Sprintf("%s-%s-%s-%s", name, utilrand.String(5), namespace, clusterID),
```
Comment (vthapar) [inline coredns/resolver/resolver_suite_test.go:306]: Need to release note this. Longer name may cause name length related failures for some deployments.

**Review by vthapar — APPROVED** [review]
Changes look good. Reviewed and tested over weekend. Great work with E2Es too +1. Need a tracking issue to update documentation, esp Troubleshooting Doc.

File: pkg/constants/constants.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ const (
 	LabelIsHeadless          = "lighthouse.submariner.io/is-headless"
 	PublishNotReadyAddresses = "lighthouse.submariner.io/publish-not-ready-addresses"
 	GlobalnetEnabled         = "lighthouse.submariner.io/globalnet-enabled"
+	KubernetesServiceName    = "kubernetes.io/service-name"
```
Comment (skitt) [inline pkg/constants/constants.go]: There’s a constant for this in the Discovery API: https://pkg.go.dev/k8s.io/api/discovery/v1#pkg-constants (`LabelServiceName`).

File: pkg/agent/controller/types.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ type ServiceImportAggregator struct {
 
 // The ServiceImportController encapsulates two resource syncers; one that watches for local cluster ServiceImports
 // from the submariner namespace and creates/updates the aggregated ServiceImport on the broker; the other that syncs
-// aggregated ServiceImports from the broker to the local service namespace. It also creates an EndpointController.
+// aggregated ServiceImports from the broker to the local service namespace. It also creates an ServiceEndpointSliceController.
```
Comment (skitt) [inline pkg/agent/controller/types.go]:
```suggestion
// aggregated ServiceImports from the broker to the local service namespace. It also creates a ServiceEndpointSliceController.
```

**Review by skitt — APPROVED** [review]
This is great! It can go in as-is, I have a couple of minor tweaks if you re-spin.

File: pkg/constants/constants.go (hunk 2 above)
Comment (tpantelis) [reply pkg/constants/constants.go]: Ahh thanks.

File: coredns/resolver/resolver_suite_test.go (hunk 1 above)
Comment (tpantelis) [reply coredns/resolver/resolver_suite_test.go:306]: yeah I thought about that. I think we should use the `GenerateName` functionality  that K8s uses for EndpointSlices with the service name as the prefix. I'm planning to do that next in a followup PR. It's going to require supporting changes in **admiral**.

**Comment by dfarrell07** [comment]
> This branch cannot be rebased due to too many changes

I've never seen that before. It looks like there was some recent weirdness along these lines:

https://github.com/orgs/community/discussions/51612

**Pull request merged**